custom_components/unifiprotect/number.py

```diff
@@ -112,23 +124,18 @@ def __init__(
         protect_data,
         server_info,
         device_id,
-        number_entity,
+        description: UnifiProtectNumberEntityDescription,
     ):
         """Initialize the Number Entities."""
-        super().__init__(
-            upv_object, protect_data, server_info, device_id, number_entity
-        )
-        self.upv = upv_object
-        self._number_entity = number_entity
-        number_item = NUMBER_TYPES[self._number_entity]
-        self._name = f"{number_item[_NUMBER_NAME]} {self._device_data['name']}"
-        self._icon = f"mdi:{number_item[_NUMBER_ICON]}"
-        self._device_type = number_item[_NUMBER_TYPE]
-        self._attr_max_value = number_item[_NUMBER_MAX_VALUE]
-        self._attr_min_value = number_item[_NUMBER_MIN_VALUE]
-        self._attr_step = number_item[_NUMBER_STEP]
-        self._attr_mode = number_item[_NUMBER_MODE]
-        self._attr_entity_category = ENTITY_CATEGORY_CONFIG
+        super().__init__(upv_object, protect_data, server_info, device_id, description)
+        self.entity_description = description
+        self._name = f"{self.entity_description.name} {self._device_data['name']}"
+        self._icon = self.entity_description.icon
+        self._device_type = self.entity_description.ufp_device_type
+        self._attr_max_value = self.entity_description.ufp_max
+        self._attr_min_value = self.entity_description.ufp_min
+        self._attr_step = self.entity_description.ufp_step
+        self._attr_entity_category = self.entity_description.entity_category
```

Comment:
Here are the attributes that can be dropped if they are provided by `self.entity_description`

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L336